### PR TITLE
Ticket_1197 and Ticket_1364

### PR DESCRIPTION
**Pull request by gpinkley**
Ticket_1197 Ticket_1364 Separated the widget for EventSearchAssetHistory into AssetHistoryStats and AssetHistoryTable. Implemented gemstone/react-table into both. Added AssetName to AssetHistoryTable

### CODE REVIEW

File: SEBrowser/Scripts/TSX/Components/EventSearch/EventSearchPreview/AssetHistoryTable.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            contentType: "application/json; charset=utf-8",
+            dataType: 'json',
+            cache: true,
+            async: true
+        });
+    }
+
+    return (
+        <div className="card">
+            <div className="card-header">Asset History Table for {assetName}:
```
Comment (elwills) [inline SEBrowser/Scripts/TSX/Components/EventSearch/EventSearchPreview/AssetHistoryTable.tsx:38]: Change this to `Event History for {assetName}:`

File: SEBrowser/Scripts/TSX/Components/EventSearch/EventSearchPreview/AssetHistoryTable.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                <select className='pull-right' value={count} onChange={(evt) => setCount(parseInt(evt.target.value))}>
```
Comment (elwills) [inline SEBrowser/Scripts/TSX/Components/EventSearch/EventSearchPreview/AssetHistoryTable.tsx:39]: Add a label to this dropdown that clarifies what it's for (i.e., `Num of Events`).

File: SEBrowser/Scripts/TSX/Components/EventSearch/EventSearchPreview/EventSearchPreviewPane.tsx [hunk 3]
```diff
@@ -144,8 +145,6 @@ export default function EventPreviewPane(props: IProps) {
 
                         else if (widget.Name === 'EventSearchFileInfo')
                             return <EventSearchFileInfo key={index} eventID={props.EventID} />;
-                        else if (widget.Name === 'EventSearchHistory')
-                            return <EventSearchHistory key={index} eventID={props.EventID} />;
```
Comment (elwills) [inline SEBrowser/Scripts/TSX/Components/EventSearch/EventSearchPreview/EventSearchPreviewPane.tsx:148]: If we're deleting this widget here, the file should probably be removed as well.

**Review by elwills — CHANGES_REQUESTED** [review]

File: SEBrowser/Controllers/OpenXDAController.cs [hunk 4]
```diff
@@ -448,9 +448,11 @@ public DataTable GetEventSearchHistory(int eventID, int count = 10)
                         TOP " + count.ToString() + @"
 	                    EventType.Name as EventType,
 	                    Event.StartTime,
-	                    Event.ID
+	                    Event.ID,
+                        Asset.AssetName
                     FROM
 	                    Event JOIN
+                        Event ON Event.AssetID = Asset.ID JOIN
```
[the remark this answers is not in the document]
Comment (gpinkley) [reply SEBrowser/Controllers/OpenXDAController.cs]: Done

File: SEBrowser/Scripts/TSX/Components/EventSearch/EventSearchPreview/AssetHistoryStats.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+            cache: true,
+            async: true
+        });
+    }
+
+    let tableData = Object.keys(statsData).map((key) => ({ Stat: key, Value: statsData[key] }));
+
+    return (
+        <div className="card">
+            <div className="card-header">Asset History Stat:</div>
```
[the remark this answers is not in the document]
Comment (gpinkley) [reply SEBrowser/Scripts/TSX/Components/EventSearch/EventSearchPreview/AssetHistoryStats.tsx]: Done

File: SEBrowser/Scripts/TSX/Components/EventSearch/EventSearchPreview/AssetHistoryTable.tsx (hunk 1 above)
Comment (gpinkley) [reply SEBrowser/Scripts/TSX/Components/EventSearch/EventSearchPreview/AssetHistoryTable.tsx]: Done

File: SEBrowser/Scripts/TSX/Components/EventSearch/EventSearchPreview/AssetHistoryTable.tsx (hunk 2 above)
Comment (gpinkley) [reply SEBrowser/Scripts/TSX/Components/EventSearch/EventSearchPreview/AssetHistoryTable.tsx]: Done

File: SEBrowser/Scripts/TSX/Components/EventSearch/EventSearchPreview/EventSearchPreviewPane.tsx (hunk 3 above)
Comment (gpinkley) [reply SEBrowser/Scripts/TSX/Components/EventSearch/EventSearchPreview/EventSearchPreviewPane.tsx:148]: Done

File: SEBrowser/Controllers/OpenXDAController.cs [hunk 6]
```diff
@@ -26,6 +26,7 @@
 using GSF.Data;
 using GSF.Data.Model;
 using GSF.Web;
+using Microsoft.Ajax.Utilities;
```
Comment (clackner-gpa) [inline SEBrowser/Controllers/OpenXDAController.cs:29]:
```suggestion
```

I don't think we need this

File: SEBrowser/Controllers/OpenXDAController.cs [hunk 7]
```diff
@@ -488,12 +491,11 @@ Asset  JOIN
 	                    (SELECT ROUND(MAX(IMax),3) as IMax FROM (VALUES(IAMax), (IBMax), (ICMax)) AS IMaxView(IMax)) as IMax OUTER APPLY
 	                    (SELECT ROUND(MAX(I2tMax),3) as I2tMax FROM (VALUES(IA2t), (IB2t), (IC2t)) AS I2tView(I2tMax)) as I2tMax
                     WHERE Asset.ID = (SELECT AssetID FROM Event WHERE ID = {0})
-                    GROUP BY VoltageKV
+                    GROUP BY VoltageKV , Asset.AssetName
```
Comment (clackner-gpa) [inline SEBrowser/Controllers/OpenXDAController.cs:494]:
```suggestion
                    GROUP BY Asset.AssetName

```

File: SEBrowser/Scripts/TSX/Components/EventSearch/EventSearchPreview/AssetHistoryTable.tsx [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+const AssetHistoryTable: React.FC<SEBrowser.IWidget<any>> = (props) => {
+    const [historyData, setHistoryData] = React.useState<Array<any>>([]);
+    const [count, setCount] = React.useState<number>(10);
+    const [assetName, setAssetName] = React.useState<string>('');
+
+    React.useEffect(() => {
+        let handle = getHistoryData();
+        handle.done((data) => {
+            console.log(data);
```
Comment (clackner-gpa) [inline SEBrowser/Scripts/TSX/Components/EventSearch/EventSearchPreview/AssetHistoryTable.tsx:14]:
```suggestion
```

remove all `console.log` unless they are absolutely necessary

**Review by clackner-gpa — CHANGES_REQUESTED** [review]

**Comment by clackner-gpa** [comment]
@gpinkley We may have talked about that as part of the SOE Engine but I would swap the controller to return a `DataRow` if it only needs the first entry.

File: SEBrowser/Controllers/OpenXDAController.cs (hunk 7 above)
Comment (gpinkley) [reply SEBrowser/Controllers/OpenXDAController.cs:494]: Making this change causes the following error: System.Data.SqlClient.SqlException: 'Column 'Asset.VoltageKV' is invalid in the select list because it is not contained in either an aggregate function or the GROUP BY clause.'

**Review by clackner-gpa — APPROVED** [review]

**Review by elwills — APPROVED** [review]

**Comment by elwills** [comment]
The SQL script for XDA will need to be updated to load the correct widgets: https://github.com/GridProtectionAlliance/openXDA/blob/master/Source/Data/11%20-%20SEBrowser.sql#L231

**Review by elwills — APPROVED** [review]
